lib/waistband/model.rb

```diff
@@ -0,0 +1,195 @@
+require 'digest/md5'
+require 'active_support/core_ext/object/blank'
+require 'active_model'
+
+module Waistband
+  class Model
+
+    extend ::ActiveModel::Callbacks
+
+    include ::ActiveModel::Validations
+    include ::ActiveModel::Validations::Callbacks
+
+
+    #################
+    # Class methods #
+    #################
+
+    class << self
+
+      attr_reader :type_name, :column_defaults
+
+      def index_name(name)
+        @index_name = name.to_s
+      end
+
+      def index_type(name)
+        @type_name = name.to_s
+      end
+
+      def index
```

Comment:
Maybe the dsl explicitly loads a yml? Maybe you just stick with the naming convention you have...

```
index do
  config Rails.root.join('/config/model.yml')
end
```